File: container2/coding_problems/easy/general/4/solution.js
const fibo = { 1: 0, 2: 1}

function getNthFib(n) {
  if (fibo.hasOwnProperty(n)) return fibo[n]

  if (fibo[n-1]) return fibo[n-1] + fibo[n-2]


  fibo[n-1] = getNthFib(n-1)
	fibo[n-2] = getNthFib(n-2)
	fibo[n] = fibo[n-1] + fibo[n-2]
	return fibo[n]
}

File: container2/coding_problems/easy/general/4/solution.py
fibo = { 1: 0, 2: 1 }

def getNthFib(n):
  if n in fibo:
    return fibo[n]
  if n-1 in fibo:
    return fibo[n-1] + fibo[n-2]
  fibo[n-2] = getNthFib(n-2)
  fibo[n-1] = getNthFib(n-1)
  fibo[n] = getNthFib(n)
  return fibo[n]
